src/js/script.js

```diff
@@ -185,11 +187,92 @@ const select = {
         }
       }
     }
+
+    // multiply price by amount
+    price *= thisProduct.amountWidget.value;
+
     // update calculated price in the HTML
     thisProduct.priceElem.innerHTML = price;
     }
+
+    initAmountWidget(){
+      const thisProduct = this;
+
+      thisProduct.amountWidget = new AmountWidget(thisProduct.amountWidgetElem);
+      thisProduct.amountWidgetElem.addEventListener('updated',function(){
+        thisProduct.processOrder();
+      });
+    }
   }
   
+  class AmountWidget{
+    construtor(element){
+      const thisWidget = this;
+      console.log('AmountWidget', thisWidget);
+      console.log('constructor arguments:', element);
+
+      thisWidget.getElements(element);
+      thisWidget.setValue(thisWidget.input.value);
```

Comment:
W tym momencie `thisWidget.input` jest jeszcze `undefined`, więc nie możesz dorwać się do właściwości `value`,  dlatego w inpucie wyświetla `NaN` (wynik wykonania `parseInt(undefined)`). O ile dobrze rozumiem treść zadania jako wartości początkowej powinnaś użyć `settings.amountWidget.defaultValue`
```suggestion
      thisWidget.setValue(settings.amountWidget.defaultValue);
```